#! /bin/sh /usr/share/dpatch/dpatch-run
## 08-local-getline.dpatch by  <bdefreese@bddebian3.bddebian.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Getline now supplied by glibc, rename local getline to local_getline.

@DPATCH@
diff -urNad thttpd-2.25b~/extras/htpasswd.c thttpd-2.25b/extras/htpasswd.c
--- thttpd-2.25b~/extras/htpasswd.c	2009-10-29 23:09:05.000000000 -0400
+++ thttpd-2.25b/extras/htpasswd.c	2009-10-29 23:09:44.000000000 -0400
@@ -54,7 +54,7 @@
     while((line[y++] = line[x++]));
 }
 
-static int getline(char *s, int n, FILE *f) {
+static int local_getline(char *s, int n, FILE *f) {
     register int i=0;
 
     while(1) {
@@ -225,7 +225,7 @@
 
     found = 0;
     /* line we get is username:pwd, or possibly any other cruft */
-    while(!(getline(line,MAX_LINE_LEN,f))) {
+    while(!(local_getline(line,MAX_LINE_LEN,f))) {
         char *i;
 	
         if(found || (line[0] == '#') || (!line[0])) {
